_rules/first-focusable-elements-are-links-to-content-e53727.md

```diff
@@ -0,0 +1,868 @@
+---
+id: e53727
+name: First focusable elements are links to sections of content
+rule_type: atomic
+description: |
+  This rule checks that the first focusable elements are links referring to sections of content on the same page
+accessibility_requirements:
+  wcag-technique:G124: # Adding links at the top of the page to each area of the content
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS styling
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+    - Christina Adams
+  assets:
+    - _The Three Kingdoms_ by Yu Sumei (Tuttle publishing, May 2014, ISBN 9780804843935)
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectation 1
+
+Within the test target, there is an [initial segment][] of the [focusable][] elements (in focus order) such that each element in that [initial segment][]:
+
+- is [included in the accessibility tree][]; and
+- is [visible][] when [focused][]; and
+- has a [semantic role][] of link; and
+- can be activated by use of keyboard only; and
+- when activated, moves focus to a [section of content][] within the same [document][]; and
+- has an [accessible name][] that communicates that it links to that specific [section of content][].
+
+**Note:** There is no requirement on how many [focusable][] elements are part of that [initial segment][], nor any requirement to provide a way to determine (programmatically or not) where that [initial segment][] stops. Technique [G124: Adding links at the top of the page to each area of the content](https://www.w3.org/WAI/WCAG21/Techniques/general/G124), and this rule, only require that such a set exists.
+
+**Note for reviewers**: An attempt to clarify what this initial segment has to be is done in the second expectation: it must contain exactly one link for each section of content in the page. I am still not very happy with the formulation, nor with the order of these two expectations. Any suggestions to improve that are welcome…
+
+## Expectation 2
+
+Each [section of content][] in the [document][] that is preceded (in tree order) by at least one [section of repeated content][] is the target of exactly one link from the [initial segment][] of [focusable][] elements found by Expectation 1.
```

Comment:
I (now) agree that to pass G124 every section needs to be linked (the technique's name mentions "each area of the content").
But you do raise a valid concern. A segment that does not link to each section would fail this rule, but if it includes a link to the main section, and that is not the first link in the page, then it would (potentially) fail all rules...
Should we add an expectation to this rule that one of the links is to the main section?

Reply:
I've boosted up the rule, using the new def of "semantic segmentation". That should cover these cases (check Passed Examples 8-9 in main rule).